### PR TITLE
Mz/refactor decryption

### PR DESCRIPTION
**Pull request by mayeul-zama**

### CODE REVIEW

**Comment by mayeul-zama** [comment]
@slab-ci cpu_fast_test

**Comment by IceTDrinker** [comment]
is this a task that was discussed in a weekly, i.e. is required to have soon ?

File: tfhe/docs/fine_grained_api/shortint/operations.md [hunk 1]
```diff
@@ -72,8 +72,8 @@ fn main() {
     server_key.unchecked_mul_lsb_assign(&mut ct_1, &ct_2);
 
     // We use the client key to decrypt the output of the circuit:
-    let output = client_key.decrypt(&ct_1);
-    println!("expected {}, found {}", ((msg1 * scalar as u64 - msg2) * msg2) % modulus as u64, output);
+    let output = client_key.decrypt_decode_padding(&ct_1);
```
Comment (tmontaigu) [inline tfhe/docs/fine_grained_api/shortint/operations.md:75]: I think we should keep the simple `decrypt` that returns the msg as we previously did, as I think this is what users (and us) care about most of the time

Comment (IceTDrinker) [reply tfhe/docs/fine_grained_api/shortint/operations.md:75]: agreed for most of our code, haven't had time to review all this yet

File: tfhe/docs/fine_grained_api/quick_start.md [hunk 2]
```diff
@@ -124,8 +124,8 @@ fn main() {
     let ct_3 = server_key.unchecked_add(&ct_1, &ct_2);
 
     // We use the client key to decrypt the output of the circuit:
-    let output = client_key.decrypt(&ct_3);
-    assert_eq!(output, (msg1 + msg2) % modulus as u64);
+    let output = client_key.decrypt_decode_padding(&ct_3);
```
Comment (IceTDrinker) [inline tfhe/docs/fine_grained_api/quick_start.md:127]: as indicated by Thomas for documentation I would keep the plain old decrypt as users should be using that in about all cases, applies to all md files and docstrings not related to decryption

**Review by IceTDrinker — COMMENTED** [review]
I believe we should not advertise anything else than decrypt for now in .md files and docstrings and doctests (except for doctests linked to the new decryption primitives) I would review once those changes are reverted to avoid noise

File: tfhe/src/shortint/engine/mod.rs [hunk 3]
```diff
@@ -114,6 +114,9 @@ where
             .iter_mut()
             .for_each(|a| {
                 let f_eval = f(i as u64);
+
+                assert!(f_eval < modulus_sup as u64);
```
[the remark this answers is not in the document]
Comment (mayeul-zama) [reply tfhe/src/shortint/engine/mod.rs]: Removed

**Pull request closed without merging**